src/main/java/seedu/address/model/Model.java

```diff
@@ -81,6 +82,12 @@
      */
     void addProcedure(Procedure procedure);
 
+
+    /**
+     * Calculates cost on the given date.
+     */
+    void calculateCost(Date date);
```

Comment:
May I ask the use of this method? It seems that this method isn't really used for your command.

Reply:
ah I was considering 2 different implementations, 

1. making the model run the calculation and then return the value. 

2. requesting the list and then calculating the value within the calculateCommand class. 

I think that since we are going with the first one, I can remove this method!